src/styles/main.scss

```diff
@@ -1,3 +1,67 @@
 body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+}
+
+$cell-width: 100px;
+$cell-height: 100px;
+$border-size: 1px;
+$gap: 1px;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: 750px;
+  position: absolute;
+  
+  @for $index from 1 through 31 {
+    :nth-child(#{$index})::before {
+      display: flex;
+      justify-content: center;
+      height: 100px;
+      line-height: 100px;
+      content: "#{$index}";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i} {
+      :nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  $start-days:
+    "mon" 0,
+    "tue" $cell-width+1.6px+$gap,
+    "wed" 2*$cell-width+3.2px+2$gap,
+    "thu" 3*$cell-width+4.8px+3$gap,
+    "fri" 4*$cell-width+6.4px+4$gap,
+    "sut" 5*$cell-width+8px+5$gap,
+    "sun" 6*$cell-width+9.6px+6$gap;
+
+  @each $name, $start-day in $start-days {
+    &--day-#{$name} :first-child {
+      margin-left: #{$start-day};
+    }
+  }
+
+  &__day {
+    width: #{$cell-width};
+    height: #{$cell-height};
+    background-color: #eee;
+    border: #{$border-size} solid #000;
+    transition-duration: 0.5s;
```

Comment:
Also specify the properties you change on hover

Reply:
comment wasn't fixed. add transition-property